rfcs/2022-08-22-11563-component-demo-mode.md

```diff
@@ -0,0 +1,163 @@
+# RFC 11563 - 2022-08-22 - Component Demo mode
+
+We would like the ability to demo any Vector configuration without external
+dependencies. This would greatly reduce the friction of demoing for the
+purposes of education, sales, and development of tools that depend on Vector
+(i.e., dashboards, a UI, etc.).
+
+## Scope
+
+### In scope
```

Comment:
If demo mode is turned on for a source, that source will generate data to be sent upstream rather than rely on input from an external source. For a sink, it will just prevent the sink from sending data on to the external source. `vector tap` can be used to inspect the data that is being received by the sink. 

Say we have Source A in demo mode and Source B not in demo mode, both connected to Sink C - Sink C will receive demo data from Source A but live data from Source B.